main.tf

```diff
@@ -122,12 +122,11 @@ resource "kubernetes_job" "db_init_job" {
           command = [
             "/bin/sh",
             "-c",
-            "psql $DATABASE_URL -c \"CREATE DATABASE ${each.key};\""
+            "psql $DATABASE_URL -tc \"SELECT 1 FROM pg_database WHERE datname = '${each.key}'\" | grep -q 1 || psql $DATABASE_URL -c \"CREATE DATABASE ${each.key};\""
```

Comment:
hmm I don't know if we want to do this. 

I was thinking we should just make the database creation be a dependency of the materialize_instance creation... IE if the job fails we can't create the environment.

I could be wrong about this, happy to discuss.

Reply:
Oh yes, I see what you mean, if we were to block the Materialize CRD creation if the db creation fails it would make sense as people will not accidentally reuse databases for existing instances that would have already written down a bunch of stuff.

Testing this out now to see if the Kubernetes Terraform provider would allow us to do that out of the box.

Reply:
Added `wait_for_completion` to the job and tested it, this allows us to fail early and not get to the CRD creation at all.

If the db exists users could decide to set the `create_database` flag to `false` and skip the job creation all together.